bftengine/src/bftengine/DbCheckpointManager.cpp

```diff
@@ -133,7 +133,7 @@ void DbCheckpointManager::cleanUp() {
   // this gets called when db checkpoint is disabled
   // check if there is chkpt data in persistence
   loadCheckpointDataFromPersistence();
-  if (!maxNumOfCheckpoints_) {
+  if (!maxNumOfCheckpoints_ || !ReplicaConfig::instance().dbCheckpointFeatureEnabled) {
```

Comment:
nit: 
its better to keep the order like this:
if ( !ReplicaConfig::instance().dbCheckpointFeatureEnabled || !maxNumOfCheckpoints_ )